src/Datadog.Trace/SpanContextPropagator.cs

```diff
@@ -66,8 +72,9 @@ public SpanContext Extract(IHeadersCollection headers)
 
             var parentId = ParseUInt64(headers, HttpHeaderNames.ParentId);
             var samplingPriority = ParseEnum<SamplingPriority>(headers, HttpHeaderNames.SamplingPriority);
+            var origin = ParseString(headers, HttpHeaderNames.Origin);
```

Comment:
There may be another edge case here. Are we guaranteed to have an incoming Trace ID in the headers? Earlier in the method, if there is no Trace ID in the headers, we will skip generating a new SpanContext altogether and the Origin will not be recorded. If we have to handle this case, maybe we can work together on the solution because I'm going to be working in the same area to turn a subset of headers into tags on root spans

Reply:
That case affects not only the `origin` header but also the `sampling` header, in this particular case the `origin` is used also for `sampling` decisions, that´s the reason to treat it the same way as the `sampling` header in this PR. I don´t know what is the behavior we want in case of a null/invalid Trace ID.

Reply:
According to the RFC, Synthetics will start a distributed trace and will send trace id and parent id headers.

Reply:
Thanks for the clarification @lucaspimentel , looks like this is not a concern then. Resolving